packages/kbn-std/src/observables.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Observable } from 'rxjs';
+import { first, last, take, takeLast, toArray } from 'rxjs/operators';
+
+export function firstValueFrom<T>(observable: Observable<T>): Promise<T> {
+  // type casting required for rxjs v7
+  return observable.pipe(first()).toPromise() as Promise<T>;
```

Comment:
will it be `Promise<T>`? why `optionalFirstValueFrom` is not provided in rxjs v7?

Reply:
Yeah, `first()` throws an error if the observable is empty, though I'm tempted to implement these without operators to be a little more clear

Reply:
I'm not sure why RxJS doesn't include optional helpers, but I'm assuming that they want people to just use `defaultIfEmpty(undefined)` or something?

Reply:
Just guessing, but I would think emitting empty values from an observable is an anti-pattern, so you should rather fix your source observable.

Reply:
I don't think an observable which doesn't emit any values is broken necessarily.

Observables like this will cause the promise returned from `firstValueFrom()` to throw, but in RxJS 6, using `.pipe(take(1)).toPromise()` would actually resolve the promise with undefined. We use `take()` all over the place in Kibana and I think it's important that we don't just wholesale convert those to `firstValueFrom()` and assume that the observable will always produce a value, but maybe I'm wrong.

Before making this PR I got about 3/4 of the way though converting all our `.toPromise()` usage by hand, and I found a lot of places where we use `take(1)` instead of `first()`. In many places it wasn't clear if the consumer intends to allow for the observable to complete without emitting any notifications or would prefer to thrown an `EmptyError` if the source observable doesn't produce any values. It's because of this that I expect to need `optionalFirstValueFrom()` in a lot of places while migrating to RxJS 7, and why I'd like to put it in a central place.

Reply:
Alright, I did another quick scan of the repo to quickly audit `take(1)` usage and it's almost all when using the config or license services, which should always produce a value. I'll remove the optional helpers. Thanks for challenging this!